src/App/Fossa/Config/ReleaseGroup/AddProjects.hs

```diff
@@ -0,0 +1,97 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module App.Fossa.Config.ReleaseGroup.AddProjects (
+  AddProjectsOpts (..),
+  AddProjectsConfig (..),
+  subcommand,
+  cliParser,
+  mergeOpts,
+) where
+
+import App.Fossa.Config.Common (configHelp)
+import App.Fossa.Config.ConfigFile (ConfigFile, configReleaseGroup, configReleaseGroupProjects, configReleaseGroupRelease, configReleaseGroupTitle)
+import App.Fossa.Config.EnvironmentVars (EnvVars)
+import App.Fossa.Config.ReleaseGroup.Common (ReleaseGroupCommonOpts (..), ReleaseGroupProjectOpts (..), collectApiOpts, mergeReleaseGroupProjectRevision, mergeReleaseGroupRelease, mergeReleaseGroupTitle, releaseGroupCommonOpts, releaseGroupProjectOpts)
+import App.Types (ReleaseGroupReleaseRevision (..))
+import Control.Effect.Diagnostics (Diagnostics, Has)
+import Data.Aeson (ToJSON, defaultOptions, genericToEncoding, toEncoding)
+import Data.Text (Text)
+import Fossa.API.Types (ApiOpts)
+import GHC.Generics (Generic)
+import Options.Applicative (
+  CommandFields,
+  InfoMod,
+  Mod,
+  Parser,
+  command,
+  helpDoc,
+  info,
+  long,
+  many,
+  optional,
+  short,
+  strOption,
+ )
+import Options.Applicative.Builder (progDescDoc)
+import Style (applyFossaStyle, formatStringToDoc, stringToHelpDoc)
+
+releaseGroupAddProjectsInfo :: InfoMod a
+releaseGroupAddProjectsInfo = progDescDoc $ formatStringToDoc "Add FOSSA projects a FOSSA release group"
+
+subcommand :: (AddProjectsOpts -> a) -> Mod CommandFields a
+subcommand f = command "add-projects" $ info (f <$> cliParser) releaseGroupAddProjectsInfo
+
+data AddProjectsConfig = AddProjectsConfig
+  { apiOpts :: ApiOpts
+  , title :: Text
+  , releaseGroupReleaseRevision :: ReleaseGroupReleaseRevision
+  }
+  deriving (Eq, Ord, Show, Generic)
+
+instance ToJSON AddProjectsConfig where
+  toEncoding = genericToEncoding defaultOptions
+
+data AddProjectsOpts = AddProjectsOpts
+  { releaseGroupCommon :: ReleaseGroupCommonOpts
+  , configOpts :: Maybe FilePath
+  , titleOpts :: Maybe Text
+  , releaseOpts :: Maybe Text
+  , projectsOpts :: Maybe [ReleaseGroupProjectOpts]
+  }
+  deriving (Eq, Ord, Show, Generic)
+
+cliParser :: Parser AddProjectsOpts
+cliParser =
+  AddProjectsOpts
+    <$> releaseGroupCommonOpts
+    <*> optional (strOption (applyFossaStyle <> long "config" <> short 'c' <> helpDoc configHelp))
```

Comment:
I agree there is lot of shared options between our subcommands. I think it might be worth creating a separate ticket (I can create this) to break out all the shared options and put them into a common module. What do you think?

Reply:
I'm fine with a ticket, but I was only asking for sharing for the config values defined in this PR. That we can't get sharing for everything doesn't mean it isn't good to do it where it's possible and relatively easy.

Reply:
You right, extracted common cli opts to the Common module.